packages/core/src/components/modal/modal.tsx

```diff
@@ -84,6 +83,36 @@ export class TdsModal {
     }
   }
 
+  componentWillLoad() {
+    this.initializeModal();
+  }
+
+  disconnectedCallback() {
+    this.cleanupModal();
+  }
+
+  /** Initializes or re-initializes the modal, setting up event listeners. */
+  @Method()
+  async initializeModal() {
+    this.setDismissButtons();
+    this.setShowButton();
+  }
+
+  /** Cleans up event listeners and other resources. */
+  @Method()
```

Comment:
Should be marked as `@internal` if it not something a user might use

Reply:
I am thinking that maybe users would need to use it in case something goes off with the lifecycle. I can not predict the case to be frank, but I would rather leave it possible to use it.